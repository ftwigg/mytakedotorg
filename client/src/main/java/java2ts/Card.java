/*
 * MyTake.org
 *
 *  Copyright 2017 by its authors.
 *  Some rights reserved. See LICENSE, https://github.com/mytake/mytake/graphs/contributors
 */
package java2ts;

import com.jsoniter.any.Any;

@jsweet.lang.Interface
public class Card {
	public String username;
	public String titleSlug;
	public String title;
	public Any blocks;
	public int[] previewBlocks;
}
