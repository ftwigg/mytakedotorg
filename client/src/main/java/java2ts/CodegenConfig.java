/*
 * MyTake.org
 *
 *  Copyright 2017 by its authors.
 *  Some rights reserved. See LICENSE, https://github.com/mytake/mytake/graphs/contributors
 */
package java2ts;

import com.jsoniter.spi.TypeLiteral;
import com.jsoniter.static_codegen.StaticCodegenConfig;

@jsweet.lang.Erased
public class CodegenConfig implements StaticCodegenConfig {
	@Override
	public void setup() {
		// calls to JsoniterSpi
	}

	@SuppressWarnings("rawtypes")
	@Override
	public TypeLiteral[] whatToCodegen() {
		return new TypeLiteral[]{
				TypeLiteral.create(DraftRev.class),
				TypeLiteral.create(DraftPost.class),
				TypeLiteral.create(LoginCookie.class),
				TypeLiteral.create(PublishResult.class)
		};
	}
}
