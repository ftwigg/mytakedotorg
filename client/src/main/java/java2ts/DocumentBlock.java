/*
 * MyTake.org
 *
 *  Copyright 2017 by its authors.
 *  Some rights reserved. See LICENSE, https://github.com/mytakedotorg/mytakedotorg/graphs/contributors
 */
package java2ts;

@jsweet.lang.Interface
public class DocumentBlock {
	public StringTypes.document kind;
	public String excerptId;
	public jsweet.util.tuple.Tuple2<Integer, Integer> highlightedRange;
	public jsweet.util.tuple.Tuple2<Integer, Integer> viewRange;
}
