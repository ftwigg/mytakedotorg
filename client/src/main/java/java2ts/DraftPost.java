/*
 * MyTake.org
 *
 *  Copyright 2017 by its authors.
 *  Some rights reserved. See LICENSE, https://github.com/mytakedotorg/mytakedotorg/graphs/contributors
 */
package java2ts;

import com.jsoniter.any.Any;

@jsweet.lang.Interface
public class DraftPost implements Json {
	@jsweet.lang.Optional
	public DraftRev parentRev;
	public String title;
	public Any blocks;
}
