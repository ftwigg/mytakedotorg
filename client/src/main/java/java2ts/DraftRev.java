/*
 * MyTake.org
 *
 *  Copyright 2017 by its authors.
 *  Some rights reserved. See LICENSE, https://github.com/mytake/mytake/graphs/contributors
 */
package java2ts;

@jsweet.lang.Interface
public class DraftRev implements Json {
	public int draftid;
	public int lastrevid;
}
