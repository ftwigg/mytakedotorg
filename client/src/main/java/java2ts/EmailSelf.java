/*
 * MyTake.org
 *
 *  Copyright 2017 by its authors.
 *  Some rights reserved. See LICENSE, https://github.com/mytakedotorg/mytakedotorg/graphs/contributors
 */
package java2ts;

@jsweet.lang.Interface
public class EmailSelf implements Json {
	public String subject;
	public String body;
}
