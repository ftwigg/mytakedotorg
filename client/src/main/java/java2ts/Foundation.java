/*
 * MyTake.org
 *
 *  Copyright 2017 by its authors.
 *  Some rights reserved. See LICENSE, https://github.com/mytake/mytake/graphs/contributors
 */
package java2ts;

import java.util.Date;
import java.util.List;
import java2ts.StringTypes.recorded;
import jsweet.util.tuple.Tuple2;

public interface Foundation {
	///////////////////////////////
	// Unused, but what I'd like //
	///////////////////////////////
	@jsweet.lang.Interface
	public class Fact {
		public String title;
		public String primaryDate; // yyyy-mm-dd
		public String primaryDateKind;
		public String kind;
	}

	@jsweet.lang.Interface
	public class FactLink {
		public Fact fact;
		public String hash;
	}

	@jsweet.lang.Interface
	public class FactContent {
		public Fact fact;
	}

	@jsweet.lang.Interface
	public class VideoFactContent extends FactContent {
		public String youtubeId;
		@jsweet.lang.Optional
		public List<Person> speakers;
		@jsweet.lang.Optional
		public List<CaptionWord> transcript;
		@jsweet.lang.Optional
		public List<SpeakerMap> speakerMap;
	}

	@jsweet.lang.Interface
	public class DocumentFactContent extends FactContent {
		public List<DocumentComponent> components;
	}

	@jsweet.lang.Interface
	public class DocumentComponent {
		public String component;
		public String innerHTML;
		public int offset;
	}

	//////////////////
	// What we have //
	//////////////////
	@jsweet.lang.Interface
	public class DocumentFact {
		public String title;
		public String filename;
		public Date primaryDate;
		// TODO: https://github.com/cincheo/jsweet/issues/412
		public String primaryDateKind;
	}

	@jsweet.lang.Interface
	public class VideoFact {
		public String id;
		public String title;
		public Date primaryDate;
		public recorded primaryDateKind;
	}

	@jsweet.lang.Interface
	public class Person {
		public String firstname;
		@jsweet.lang.Optional
		public String middlename;
		public String lastname;
	}

	@jsweet.lang.Interface
	public class SpeakerMap {
		public String speaker; // lastname of Person
		public Tuple2<Integer, Integer> range;
	}

	@jsweet.lang.Interface
	public class CaptionWord {
		public int idx;
		public String word;
		public double timestamp;
	}

	@jsweet.lang.Interface
	public class CaptionMeta {
		public List<Person> speakers;
		public List<SpeakerMap> speakerMap;
	}
}
