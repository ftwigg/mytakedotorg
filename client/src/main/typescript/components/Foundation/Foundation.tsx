import * as React from "react";
import * as ReactDOM from "react-dom";
import TimelineView from "../TimelineView";

interface FoundationProps {}

class Foundation extends React.Component<FoundationProps, {}> {
  constructor(props: FoundationProps) {
    super(props);
  }
  render() {
    return (
      <div className="foundation">
        <div className="foundation__inner-container">
          <h1 className="foundation__heading">Foundation of Facts</h1>
          <p className="timeline__instructions">
            Explore Facts in the timeline below.
          </p>
        </div>
        <TimelineView />
      </div>
    );
  }
}

export default Foundation;
