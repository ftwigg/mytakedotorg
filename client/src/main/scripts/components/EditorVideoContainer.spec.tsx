import * as React from "react";
import * as renderer from "react-test-renderer";
import {} from "jest";
import {
  EditorVideoBranch,
  EditorVideoContainerProps,
  EditorVideoContainerState
} from "./EditorVideoContainer";
import { Foundation } from "../java2ts/Foundation";

const mockFn = jest.fn();

jest.mock("./Video", () => ({
  default: "Video"
}));

const containerProps: EditorVideoContainerProps = {
  idx: 2,
  active: false,
  block: {
    kind: "video",
    range: [2, 4],
    videoId: "U8MV5KDDaxumxZOCJOzExAUAAkSoYNhycVXq7jZ59_0="
  }
};

const videoFact: Foundation.VideoFactContent = {
  youtubeId: "ApTLB76Nmdg",
  speakers: [
    {
      firstname: "Martha",
      middlename: null,
      lastname: "Raddatz"
    }
  ],
  transcript: [
    {
      idx: 0,
      word: "Good ",
      timestamp: 0.75
    },
    {
      idx: 1,
      word: "evening ",
      timestamp: 0.96
    },
    {
      idx: 2,
      word: "I'm ",
      timestamp: 1.56
    },
    {
      idx: 3,
      word: "Martha ",
      timestamp: 1.92
    },
    {
      idx: 4,
      word: "Raddatz ",
      timestamp: 2.129
    },
    {
      idx: 5,
      word: "from ",
      timestamp: 2.52
    },
    {
      idx: 6,
      word: "ABC ",
      timestamp: 3.06
    },
    {
      idx: 7,
      word: "News. ",
      timestamp: 3.09
    }
  ],
  fact: {
    title: "Donald Trump - Hillary Clinton (2/3)",
    primaryDate: "2016-10-09",
    primaryDateKind: "recorded",
    kind: "video"
  }
};

test("Video loading", () => {
  const containerState: EditorVideoContainerState = {
    loading: true,
    error: false
  };

  const tree = renderer
    .create(
      <EditorVideoBranch
        containerProps={containerProps}
        containerState={containerState}
        handleRetryClick={mockFn}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("Error loading Video", () => {
  const containerState: EditorVideoContainerState = {
    loading: false,
    error: true
  };

  const tree = renderer
    .create(
      <EditorVideoBranch
        containerProps={containerProps}
        containerState={containerState}
        handleRetryClick={mockFn}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("Successfully loaded Video", () => {
  const containerState: EditorVideoContainerState = {
    loading: false,
    error: false,
    videoFact: videoFact
  };

  const tree = renderer
    .create(
      <EditorVideoBranch
        containerProps={containerProps}
        containerState={containerState}
        handleRetryClick={mockFn}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
