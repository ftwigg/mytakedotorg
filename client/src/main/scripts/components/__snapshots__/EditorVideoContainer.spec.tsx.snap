// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error loading Video 1`] = `
<div
  className="editor__document editor__document--base editor__document--hover"
>
  <h2
    className="editor__document-title"
  >
    Error loading Foundation Video
  </h2>
  <button
    className="editor__button editor__button--reload"
    onClick={[Function]}
  >
    retry?
  </button>
</div>
`;

exports[`Successfully loaded Video 1`] = `
<div
  className="editor__video-container"
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  tabIndex={0}
>
  <Video
    onSetClick={[Function]}
    timeRange={
      Array [
        2,
        4,
      ]
    }
    videoFact={
      Object {
        "fact": Object {
          "kind": "video",
          "primaryDate": "2016-10-09",
          "primaryDateKind": "recorded",
          "title": "Donald Trump - Hillary Clinton (2/3)",
        },
        "speakers": Array [
          Object {
            "firstname": "Martha",
            "lastname": "Raddatz",
            "middlename": null,
          },
        ],
        "transcript": Array [
          Object {
            "idx": 0,
            "timestamp": 0.75,
            "word": "Good ",
          },
          Object {
            "idx": 1,
            "timestamp": 0.96,
            "word": "evening ",
          },
          Object {
            "idx": 2,
            "timestamp": 1.56,
            "word": "I'm ",
          },
          Object {
            "idx": 3,
            "timestamp": 1.92,
            "word": "Martha ",
          },
          Object {
            "idx": 4,
            "timestamp": 2.129,
            "word": "Raddatz ",
          },
          Object {
            "idx": 5,
            "timestamp": 2.52,
            "word": "from ",
          },
          Object {
            "idx": 6,
            "timestamp": 3.06,
            "word": "ABC ",
          },
          Object {
            "idx": 7,
            "timestamp": 3.09,
            "word": "News. ",
          },
        ],
        "youtubeId": "ApTLB76Nmdg",
      }
    }
  />
</div>
`;

exports[`Video loading 1`] = `
<div
  className="editor__document editor__document--base editor__document--hover"
>
  <h2
    className="editor__document-title"
  >
    Loading
  </h2>
</div>
`;
